examples/squiral/main.go

```
@@ -0,0 +1,341 @@
+			set := true
+			for _, out := range outers[horivert] {
+				xtarg.x = target.x + out.x
+				xtarg.y = target.y + out.y
+
+				// If one of the outer targets equals the squiral's
+				// color, again continue with next direction.
+				if auto.colorMap[xtarg.x][xtarg.y] == s.col {
+					// If this is not free we cannot go into this direction.
+					set = false
+					break // try next direction
+				}
+
+				xtarg.x = ntarg.x + out.x
+				xtarg.y = ntarg.y + out.y
+
+				// If one of the outer targets equals the squiral's
+				// color, again continue with next direction.
+				if auto.colorMap[xtarg.x][xtarg.y] == s.col {
+					// If this is not free we cannot go into this direction.
+					set = false
+					break // try next direction
+				}
+			}
+
+			if set {
+				s.pos = target
+				s.dir = dir
+				// 2. set the color of this squiral to its
+				// current position.
+				setpix(s.pos, s.col)
+				return
+			}
+		}
+	}
+
+	s.dead = true
+}
+
+type automaton struct {
+	squirals [numOfSquirals]squiral
+	colorMap [width][height]color.Color
+}
+
+func (au *automaton) init() {
+	// Init the test grid with color (0,0,0,0) and the borders of
+	// it with color(0,0,0,254) as a blocker color, so the squirals
+	// cannot escape the scene.
+	for x := 0; x < width; x++ {
+		for y := 0; y < height; y++ {
+			if x == 0 || x == width-1 || y == 0 || y == height-1 {
+				auto.colorMap[x][y] = blocker
+			} else {
+				auto.colorMap[x][y] = free
+			}
+		}
+	}
+
+	for i := 0; i < numOfSquirals; i++ {
+		auto.squirals[i].spawn()
+	}
+}
+
+func (au *automaton) step() {
+	for i := 0; i < numOfSquirals; i++ {
+		for s := 0; s < au.squirals[i].speed; s++ {
+			au.squirals[i].step(i)
+			if au.squirals[i].dead {
+				au.squirals[i].spawn()
+			}
+		}
+	}
+}
+
+func setpix(xy vec2, col color.Color) {
+	canvas.Set(xy.x, xy.y, col)
+	auto.colorMap[xy.x][xy.y] = col
+}
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+	c, err := ebiten.NewImage(width, height, ebiten.FilterNearest)
```

You can ignore the error form `NewImage`. This is always nil. This error type is just for backward compatibility :-)

Also please use `ebiten.FilterDefault`.

Then

```
canvas, _ = ebiten.NewImage(width, height, ebiten.FilterDefault)
```

is fine.